CesiumGltf/include/CesiumGltf/Model.h

```diff
@@ -117,7 +117,9 @@ struct CESIUMGLTF_API Model : public ModelSpec {
       Model& gltf,
       Node& node,
       Mesh& mesh,
+      const int meshId,
       MeshPrimitive& primitive,
+      const int primitiveId,
```

Comment:
Also please avoid `int` and use the appropriate specific size type. In this case it should either by `size_t` (if we think of these as indices into a vector) or `int32_t` (if we think of these as glTF object IDs).

Reply:
For consistency shouldn't `node` also become `nodeId`?

Reply:
If we're switching to indices everywhere, `ForEachNodeInSceneCallback` should also be updated.

Reply:
> Adding explicit index parameters is fine, but it's definitely a breaking change, so it needs to be documented as such.
> Also please avoid int and use the appropriate specific size type

Done and done

Reply:
I agree with Sean here. If we're going to add ID parameters, we should add all of them, to all the functions. The "we'll do it later when we need it" approach is weak because a) it may be our users who need it, not us, and b) if we decide to add a node ID later, that will be _another_ breaking changes for us and our users. If we're going to do a breaking change, we should try to do all of the related breaking changes that we can foresee at the same time, in order to minimize churn for our users (and ourselves).

I'm also ok with just removing all the IDs, and users can use pointer subtraction if they need them. But if we're going to add _any_, we should add _all_.

Reply:
I would be fine with removing the IDs as well, since it feels a bit redundant.

Reply:
Removing the ID was definitely the way to go. 

Tried this, and in doing so, realized how much code it reverted, when all we needed is indexes in one function. Doing pointer arithmetic is fine there.